### PR TITLE
CI: Use 'uname -m' for QMAKE_APPLE_DEVICE_ARCHS

### PR DESCRIPTION
**Pull request by SCG82**
### Description

Use `uname -m` instead of `arch` to set `QMAKE_APPLE_DEVICE_ARCHS` when building Qt.

### Motivation and Context

Currently, `QMAKE_APPLE_DEVICE_ARCHS` is set using `arch`, which returns `i386` for Intel-based machines. `QMAKE_APPLE_DEVICE_ARCHS` expects `x86_64`, which `uname -m` returns. Both `arch` and `uname -m` return `arm64` on Apple M1 machines.

### How Has This Been Tested?

With `QMAKE_APPLE_DEVICE_ARCHS` being set by `arch` (result `i386`), Qt 5.15.2 failed to build on my 2018 Mac Mini (6-core) running macOS 11.2. Setting `QMAKE_APPLE_DEVICE_ARCHS` using `uname -m` (result `x86_64`) resolved this issue. 

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: build-script-macos-02.sh [hunk 1]
```diff
@@ -200,7 +200,7 @@ build_06_build_dependency_qt() {
       -skip qtscript -skip qtscxml -skip qtsensors -skip qtserialbus -skip qtspeech \
       -skip qttranslations -skip qtwayland -skip qtwebchannel -skip qtwebengine -skip qtwebglplugin \
       -skip qtwebsockets -skip qtwebview -skip qtwinextras -skip qtx11extras -skip qtxmlpatterns \
-      QMAKE_APPLE_DEVICE_ARCHS=`arch`
+      QMAKE_APPLE_DEVICE_ARCHS="$(uname -m)"
```
Comment (RytoEX) [inline build-script-macos-02.sh:203]: Don't modify the shell scripts.  They are automatically rebuilt using the `create_scripts.yml` workflow.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: build-script-macos-02.sh (hunk 1 above)
Comment (SCG82) [reply build-script-macos-02.sh]: Modified `build_deps.yml` (and found an extra `=` in the `CXX` assignment)

File: .github/workflows/build_deps.yml [hunk 2]
```diff
@@ -229,7 +229,7 @@ jobs:
             brew install gcc || true
             CC="/usr/local/bin/gcc"
             LD="/usr/local/bin/gcc"
-            CXX=="/usr/local/bin/g++"
+            CXX="/usr/local/bin/g++"
```
Comment (RytoEX) [inline .github/workflows/build_deps.yml:232]: This CXX change should be a separate commit and/or a separate PR.  Otherwise, blame will point to this commit, and reverting this commit will revert both changes.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: .github/workflows/build_deps.yml (hunk 2 above)
Comment (SCG82) [reply .github/workflows/build_deps.yml]: done
